app/Http/Controllers/SaleController.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Sale;
+use Illuminate\Http\Request;
+use Illuminate\Validation\ValidationException;
+
+class SaleController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // create a list of possible statuses
+        $statuses = Sale::PENDING . "," . Sale::RECEIVED . "," . Sale::PAID . "," . Sale::CANCELLED;
+        $paymentTypes = Sale::VISA . "," . Sale::MASTER_CARD;
+        try {
+            // validate the data to be stored is correct
+            $request->validate([
+                'client_name' => 'required|string|max:255',
+                "status" => "required|string|in:{$statuses}",
+                "payment_type" => "required|string|in:{$paymentTypes}",
+                'card_number' => 'required|string|size:16|regex:/^[0-9]*$/',
+                'cardholder_name' => 'required|string|max:255',
+                'price' => 'required|numeric|min:0',
+                'description'=>'required|string|max:255',
+            ]);
+        } catch (ValidationException $e) {
+            return response([
+                'success' => false,
+                'errors' => $e->errors()
+            ]);
+        }
+
+        // grab the data to be added from the request body
+        $params = $request->only([
+            "client_name",
+            "status",
+            "payment_type",
+            "card_number",
+            "cardholder_name",
+            "price",
+            "description",
+        ]);
+
+        // create a new sale with the passed data
+        $sale = Sale::create($params);
+        $sale->save();
```

Comment:
I think the call to `->save()` here is redundant, `create()` already inserts into the database